yt_dlp/extractor/youtube.py

```diff
@@ -3307,7 +3306,57 @@ def _extract_heatmap(self, data):
                 'value': ('intensityScoreNormalized', {float_or_none}),
             })) or None
 
-    def _extract_comment(self, comment_renderer, parent=None):
+    def _extract_comment(self, view_model, entities, parent=None):
+        comment_entity_payload = traverse_obj(entities, (..., 'payload', 'commentEntityPayload', {dict}), get_all=False)
+        toolbar_entity_payload = traverse_obj(entities, (..., 'payload', 'engagementToolbarStateEntityPayload', {dict}), get_all=False)
+        comment_id = comment_entity_payload.get('properties').get('commentId')
+
+        info = {
+            'id': comment_id,
+            'text': try_get(comment_entity_payload, lambda x: x['properties']['content']['content'], str),
+            'like_count': parse_count(try_get(comment_entity_payload, lambda x: x['toolbar']['likeCountA11y'], str)) or 0,
+            'author_id': traverse_obj(comment_entity_payload, ('author', 'channelId', {self.ucid_or_none})),
+            'author': try_get(comment_entity_payload, lambda x: x['author']['displayName'], str),
+            'author_thumbnail': traverse_obj(comment_entity_payload, ('author', 'avatarThumbnailUrl', {url_or_none})),
+            'parent': parent or 'root',
+        }
+
+        # Timestamp is an estimate calculated from the current time and time_text
+        time_text = try_get(comment_entity_payload, lambda x: x['properties']['publishedTime'], str) or ''
+        timestamp = self._parse_time_text(time_text)
+
+        info.update({
+            # FIXME: non-standard, but we need a way of showing that it is an estimate.
+            '_time_text': time_text,
+            'timestamp': timestamp,
+        })
+
+        info['author_url'] = urljoin(
+            'https://www.youtube.com',
+            traverse_obj(comment_entity_payload,
+                         ('author',
+                          'channelCommand',
+                          'innertubeCommand',
+                          'browseEndpoint',
+                          'canonicalBaseUrl'),
+                         expected_type=str, get_all=False))
+
+        author_is_uploader = traverse_obj(comment_entity_payload, ('author', 'isCreator'))
+        if author_is_uploader is not None:
+            info['author_is_uploader'] = author_is_uploader
+
+        if toolbar_entity_payload.get('heartState') == 'TOOLBAR_HEART_STATE_HEARTED':
+            info['is_favorited'] = True
+
+        info['author_is_verified'] = traverse_obj(comment_entity_payload, ('author', 'isVerified')) == 'true'
```

Comment:
are you sure it's a string like `'true'` instead of a boolean value? I had this in my WIP branch:
```suggestion
        if traverse_obj(comment_entity_payload, ('author', 'isVerified', {bool})):
            info['author_is_verified'] = True
```

Reply:
I skipped this as well because it is the same as my comment on requested change above. Making ```info['author_is_verified'] = True``` would change the ```.info.json``` format, as far as I know all the bool-like values are strings in the set ```{'true', 'false'}```.

Reply:
look at the old method. the fields it sets are using bool

Reply:
With this change, I no longer get the ```author_is_verified``` key when the value is `false`, investigating further.

Reply:
The value coming from youtube may very well be a string. That's why I was asking. My "look at the old method" comment was in response to a comment that has been deleted regarding changing the value *we* set. You can do this if it works:

```suggestion
        if traverse_obj(comment_entity_payload, ('author', 'isVerified')) == 'true':
            info['author_is_verified'] = True
```

Reply:
Yea, sorry, I deleted a few comments to not add to the confusion, I had forgotten that json serializes the bools to strings.

Reply:
It's working correctly with your original suggestion. Double checked. Commit pushed.